ui/panels-plugin/src/plugins/line-chart/utils/focused-series.ts

```diff
@@ -0,0 +1,75 @@
+// Copyright 2021 The Perses Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import type { EChartsOption } from 'echarts';
+import { GraphSeriesValueTuple } from '@perses-ui/core';
+import { TOOLTIP_DATE_FORMAT, TOOLTIP_MAX_ITEMS } from '../tooltip/tooltip-model';
+
+export interface FocusedSeriesInfo {
+  seriesIdx: number | null;
+  datumIdx: number | null;
+  seriesName: string;
+  date: string;
+  markerColor: string;
+  x: number;
+  y: number;
+}
+
+export type FocusedSeriesArray = FocusedSeriesInfo[];
+
+export function getNearbySeries(
```

Comment:
Would be nice to get a few comments here explaining what's going on (e.g. what is `xBuffer` and `yBuffer`?)